### PR TITLE
[dev] PBIs AB#56673 & AB#56674: Add MUI Chip and Deprecate Label

### PR DESCRIPTION
**Pull request by cbrewer1689**

### CODE REVIEW

File: docs/src/dataDisplay/chip/chip.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    >
+                        {description}
+                    </Typography>
+
+                    <Heading
+                        anchorLink="address"
+                        variant="h2"
+                    >
+                        Chips
+                    </Heading>
```
Comment (groberts314) [inline docs/src/dataDisplay/chip/chip.jsx:54]: Why `anchorLink="address"`?  Should it be `anchorLink="chips"`, and then the one on Line 63 should be `anchorLink="deleteable-chips"`?

File: docs/src/dataDisplay/chip/chip.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    >
+                        <ExampleChip />
+                    </Example>
+
+                    <Heading
+                        anchorLink="address"
+                        variant="h2"
+                    >
+                        Deletable Chips
+                    </Heading>
```
Comment (groberts314) [inline docs/src/dataDisplay/chip/chip.jsx:67]: Line 63 should be `anchorLink="deleteable-chips"`

File: docs/src/dataDisplay/chip/examples/exampleChip.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+function ExampleChip() {
+    const classes = useStyles();
+
+    return (
+        <div className={classes.root}>
+            <Chip color="green" disabled label="Disabled" />
+            <Chip color="green" label="Green" />
+            <Chip color="teal" label="Teal" />
+            <Chip color="sky" label="Sky" />
+            <Chip color="teal" label="Teal" />
```
Comment (groberts314) [inline docs/src/dataDisplay/chip/examples/exampleChip.jsx:29]: Teal is in here twice?  Is there another color we are supposed to include?

File: docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const onDelete = () => {
+        console.info('You clicked the delete icon.');
+    };
+
+    return (
+        <div className={classes.root}>
+            <Chip color="green" disabled label="Disabled" onDelete={onDelete} />
+            <Chip color="teal" label="Teal" onDelete={onDelete} />
+            <Chip color="sky" label="Sky" onDelete={onDelete} />
+            <Chip color="teal" label="Teal" onDelete={onDelete} />
```
Comment (groberts314) [inline docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx:32]: Teal one in here twice too

File: docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            margin: spacing(0.5),
+        },
+    },
+}));
+
+function ExampleDeletableChip() {
+    const classes = useStyles();
+
+    const onDelete = () => {
+        console.info('You clicked the delete icon.');
```
Comment (groberts314) [inline docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx:24]: Do we need to suppress ESLINT warning about `console` usage?

File: gulpfile.babel.js [hunk 6]
```diff
@@ -25,6 +25,12 @@ const paths = {
         ],
         dest: 'core',
     },
+    types: {
+        src: [
+            'src/**/*.d.ts',
+        ],
+        dest: 'core',
+    },
```
Comment (groberts314) [inline gulpfile.babel.js:33]: Nice! 👍

File: src/dataDisplay/chip/chip.jsx [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+        root: PropTypes.string,
+    }).isRequired,
+    /**
+     * Override the default delete icon element. Shown only if `onDelete` is set.
+     */
+    deleteIcon: PropTypes.element,
+    /**
+     * If `true`, the chip should be displayed in a disabled state.
+     */
+    disabled: PropTypes.bool,
```
Comment (groberts314) [inline src/dataDisplay/chip/chip.jsx:33]: Should this be `disable` for consistency with other React CM UI components?  Does MUI use `disabled`?

File: src/dataDisplay/chip/chip.jsx [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+        {
+            [classes.colorGreen]: !disabled && color === 'green',
+            [classes.colorOrange]: !disabled && color === 'orange',
+            [classes.colorPink]: !disabled && color === 'pink',
+            [classes.colorPurple]: !disabled && color === 'purple',
+            [classes.colorRed]: !disabled && color === 'red',
+            [classes.colorRedOrange]: !disabled && color === 'redOrange',
+            [classes.colorSky]: !disabled && color === 'sky',
+            [classes.colorTeal]: !disabled && color === 'teal',
+            [classes.colorTeal]: !disabled && color === 'teal',
```
Comment (groberts314) [inline src/dataDisplay/chip/chip.jsx:195]: We don't need this twice

**Review by groberts314 — CHANGES_REQUESTED** [review]
Looks good, but a few small change requests

File: docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx (hunk 5 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx:24]: `console.info` doesn't give us nag.

File: src/dataDisplay/chip/chip.jsx (hunk 7 above)
Comment (cbrewer1689) [reply src/dataDisplay/chip/chip.jsx:33]: MUI uses disabled. I am trying to get everything in sync as best as I can with them.

File: src/dataDisplay/chip/chip.jsx (hunk 8 above)
Comment (cbrewer1689) [reply src/dataDisplay/chip/chip.jsx:195]: dang, how did this happen. You get a teal, you get a teal...

File: docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx (hunk 5 above)
Comment (groberts314) [reply docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx:24]: Nice!  Resolving.

File: src/dataDisplay/chip/chip.jsx (hunk 7 above)
Comment (groberts314) [reply src/dataDisplay/chip/chip.jsx:33]: Is that gonna cause us to rethink the rest of React CM UI (where we previously standardized on `disable`)?  Just wondering ... not necessarily arguing one way or the other. 😁

File: src/dataDisplay/chip/chip.jsx (hunk 8 above)
Comment (cbrewer1689) [reply src/dataDisplay/chip/chip.jsx]: Fixed.

File: docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx (hunk 4 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx]: Fixed.

File: docs/src/dataDisplay/chip/examples/exampleChip.jsx (hunk 3 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/chip/examples/exampleChip.jsx]: Fixed.

File: docs/src/dataDisplay/chip/chip.jsx (hunk 2 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/chip/chip.jsx]: Fixed.

File: docs/src/dataDisplay/chip/chip.jsx (hunk 1 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/chip/chip.jsx]: Fixed.

Comment (groberts314) [reply docs/src/dataDisplay/chip/chip.jsx]: Resolving

File: docs/src/dataDisplay/chip/chip.jsx (hunk 2 above)
Comment (groberts314) [reply docs/src/dataDisplay/chip/chip.jsx]: Resolving

File: docs/src/dataDisplay/chip/examples/exampleChip.jsx (hunk 3 above)
Comment (groberts314) [reply docs/src/dataDisplay/chip/examples/exampleChip.jsx]: Ah ha!  _Cyan!_
Resolving.

File: docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx (hunk 4 above)
Comment (groberts314) [reply docs/src/dataDisplay/chip/examples/exampleDeletableChip.jsx]: Resolving.

File: src/dataDisplay/chip/chip.jsx (hunk 8 above)
Comment (groberts314) [reply src/dataDisplay/chip/chip.jsx]: Resolving.

**Review by groberts314 — APPROVED** [review]

File: src/dataDisplay/chip/chip.jsx (hunk 7 above)
Comment (cbrewer1689) [reply src/dataDisplay/chip/chip.jsx:33]: yeah, probably. I think we'll go back to old `disabled` eventually.

**Pull request merged**